### DISCUSSION TITLE
Is it possible to archive an entire channel (not just a single vod)?

### DISCUSSION THREAD

**Discussion by Snuupy** [Zibbp/ganymede#59] [Q&A]
Love the software, glad I can archive some vods now that I want to keep.

I know there's the feature to watch a channel for _future_ uploads (and is currently discouraged due to #57), but I wanted to know if you could archive all vods available from a channel without having to manually enter every single vod ID. Thanks.

**Comment by Zibbp** [comment]
Hi,
I can implement a feature that works like that. What were you thinking? Add a channel to a list, similar to live channels, that gets checked daily if they have new VODs to be archived?

Maybe I can consolidate the VOD and Live stream watch feature under a "watched channels" page that will allow users to select if they want live streams, VODs, or both to be watched.

**Reply by Zibbp** [reply]
Created https://github.com/Zibbp/ganymede/issues/60 for tracking.

**Reply by Snuupy** [reply]
> What were you thinking? Add a channel to a list, similar to live channels, that gets checked daily if they have new VODs to be archived?

Exactly. 😄 And only adds to the archive of vods (because twitch will automatically remove some after a certain time period).

> Maybe I can consolidate the VOD and Live stream watch feature under a "watched channels" page that will allow users to select if they want live streams, VODs, or both to be watched.

I like that more actually - less confusion about which channels are already being watched and by which type.

**Reply by Zibbp** [reply]
This feature will be added when the new UI gets released. Here is a preview of the "edit watched channel" drawer. I tried to make it as least confusing as possible. Any suggestions? The channel this entry is for is found in table where the edit button is located which opens this drawer.
![image](https://user-images.githubusercontent.com/21207065/202878322-a3ad71a0-6b31-4d29-b8b1-c50c66880fb8.png)

**Reply by Snuupy** [reply]
I love it.

Small typo "Watched Chanel" should be Channel 😅 

Actually as a new user when I selected to archive chat I did not realize it would also convert the chat json into an mp4.

Other than that I don't have any other thoughts for now besides that I didn't like that it took several subdomains to setup (and I had to reconfigure my SSL cert to include a wildcard for `*.ganymede.domain.tld`. `api-ganymede.domain.tld` wouldn't work due to CORS rules, right?

**Reply by Zibbp** [reply]
> Actually as a new user when I selected to archive chat I did not realize it would also convert the chat json into an mp4.

I have this as default, and the only option, because Ganymede is primarily an archiving platform. I thought it would be best to have a playable chat to go along with each archive for use without Ganymede. 

> Other than that I don't have any other thoughts for now besides that I didn't like that it took several subdomains to setup (and I had to reconfigure my SSL cert to include a wildcard for *.ganymede.domain.tld. api-ganymede.domain.tld wouldn't work due to CORS rules, right?

The primary reason I went with subdomains is due to cookies. Browsers are restricting cookies more and more every year and subdomains work nicely with cookies. I'm sure there are way to configure the cookies to work with other reverse proxy domain setups but I have never looked into it.

**Reply by Snuupy** [reply]
> I have this as default, and the only option, because Ganymede is primarily an archiving platform. I thought it would be best to have a playable chat to go along with each archive for use without Ganymede.

Makes sense to not rewrite the chat playback stack.

> The primary reason I went with subdomains is due to cookies. Browsers are restricting cookies more and more every year and subdomains work nicely with cookies. I'm sure there are way to configure the cookies to work with other reverse proxy domain setups but I have never looked into it.

Ah. I was thinking about this last night, `ganymede.domain.tld/api` would work instead of `api.ganymede.domain.tld`.

**Reply by Zibbp** [reply]
Hi, this feature has been released in [v1.1.0](https://github.com/Zibbp/ganymede/releases/tag/v1.1.0). I'm going to close this discussion, feel free to post in the release discussion issue https://github.com/Zibbp/ganymede/issues/63 if you have any issues or comments.
